content/automate/ManageIQ/Cloud/VM/Retirement/StateMachines/Methods.class/__methods__/start_retirement.rb

```diff
@@ -22,6 +22,11 @@
   exit MIQ_ABORT
 end
 
+unless vm.retirement_initialized?
+  $evm.log('error', "VM has not been initialized for retirement. Aborting current State Machine.")
+  exit MIQ_ABORT
```

Comment:
@d-m-u Can you raise an exception instead of exiting, it helps when we test.

Reply:
@d-m-u @mkanoor and I discussed this and the exit is okay for now. We'll address the exit changes in a future PR.